KenticoCloud.Delivery/DeliveryClient.cs

```diff
@@ -590,5 +589,30 @@ private async Task<JObject> GetResponseContent(HttpResponseMessage httpResponseM
 
             throw new DeliveryException(httpResponseMessage, faultContent);
         }
+
+        internal IEnumerable<IQueryParameter> ExtractParameters<T>(IEnumerable<IQueryParameter> parameters = null)
+        {
+            var enhancedParameters = parameters != null
+                ? new List<IQueryParameter>(parameters)
+                : new List<IQueryParameter>();
+
+            var codename = _codeFirstModelProvider.TypeProvider.GetCodename(typeof(T));
+
+            if (!IsAlreadyInParameters(parameters) && codename != null)
+            {
+                enhancedParameters.Add(new EqualsFilter("system.type", codename));
+            }
+            return enhancedParameters;
+        }
+
+        private static bool IsAlreadyInParameters(IEnumerable<IQueryParameter> parameters)
```

Comment:
What is in the parameters? Please provide better name, something like IsTypeInParameters().